source/common/http/header_map_impl.h

```diff
@@ -25,14 +25,13 @@ namespace Http {
 #define DEFINE_INLINE_HEADER_FUNCS(name)                                                           \
 public:                                                                                            \
   const HeaderEntry* name() const override { return inline_headers_.name##_; }                     \
-  HeaderEntry* name() override {                                                                   \
-    cached_byte_size_.reset();                                                                     \
```

Comment:
We should be able to kill `refreshByteSize()` and all of the hack calls we put in, right?

Reply:
All set!!!!!!

Need advice though about naming. I did keep the manual byte size count in case we wanted to assert about the byte size in the codecs and in tests. `byteSizeInternal()` is a horrible name for this... `byteSizeManual()` with a docstring explaining usage?

Reply:
I had earlier suggested you move byteSizeInternal into the _test.cc that calls it. Now I see it's also called by the fuzz-test.

But it's always this pattern:
```
EXPECT_EQ(headers.byteSize(), headers.byteSizeInternal());
```
So I think it might be more intuitive and easy-to-name if you factored that ASSERT into the method, and called it `sanityCheckByteSize()`.  WDYT?

Reply:
Can't we just have a test utility that iterates all the headers and does the count? I don't think it needs to be in the prod code?

Reply:
Trying to reconcile both not having this method exposed and @htuch's comment about enforcing internal consistency checks inside header map rather than outside, I resolved to make it a private method and have debug checks on `HeaderMap` operations. 

Tests would still catch byte size inconsistencies with every operation, and I tested byteSize in header_map_impl_test as if I didn't know what was inside `HeaderMap`. (with the pattern `EXPECT_EQ(headers.byteSize(), 0)`. Fuzz tests shouldn't get timed out by these asssertions since we default limit header map now, and our header map impl test also will never add thousands of headers.

Reply:
OK sounds good.